clippy_utils/src/ast_utils.rs

```diff
@@ -571,3 +572,36 @@ pub fn eq_mac_args(l: &MacArgs, r: &MacArgs) -> bool {
         _ => false,
     }
 }
+
+/// Extract args from an assert-like macro.
+/// Currently working with:
+/// - `assert!`, `assert_eq!` and `assert_ne!`
+/// - `debug_assert!`, `debug_assert_eq!` and `debug_assert_ne!`
+/// For example:
+/// `assert!(expr)` will return Some([expr])
+/// `debug_assert_eq!(a, b)` will return Some([a, b])
+pub fn extract_assert_macro_args(e: &Expr) -> Option<Vec<&P<Expr>>> {
```

Comment:
This function may as well determine which macro it is and return the name of the macro. You can call `is_direct_expn_of` as the final step.

You can return an array instead of `Vec`. You can deref the `P`.

```suggestion
pub fn extract_assert_macro_args(e: &Expr) -> Option<(&'static str, [&Expr; 2])> {
```

Reply:
I'm not a big fan of doing multiple things at once. However I can return `Expr` directly.

Reply:
It's not multiple things at once if you think of it as "get the assert macro parts". We have many functions like this if you look around in this same module. If you like, you can return a struct to make the output more clear. Returning the macro name will make the util more useful in general.

Reply:
I don't agree with you. We have the macro name already before even calling this function. There is no point into returning it from this function once again.

Reply:
Well you wouldn't need to get the macro name before calling the function anymore. But it's okay to leave this as is if you prefer.

Reply:
Then I'll keep as is then.